src/main/java/com/faforever/api/data/domain/Login.java

```diff
@@ -53,8 +58,23 @@ public String getUserAgent() {
     return userAgent;
   }
 
-  @OneToOne(mappedBy = "player", fetch = FetchType.LAZY)
-  public BanInfo getBanInfo() {
-    return banInfo;
+  @OneToMany(mappedBy = "player", fetch = FetchType.EAGER)
+  // Permission is managed by BanInfo class
+  @UpdatePermission(expression = "Prefab.Role.All")
+  public List<BanInfo> getBans() {
+    if (this.bans == null) {
+      this.bans = Collections.emptyList();
+    }
```

Comment:
Initialize in constructor with empty `ArrayList` instead of having a side-effect in a getter.